### ISSUE TITLE
[BUG]: strange behaviour with rc_limit

### ISSUE THREAD

**Issue by joticajulian** [koinos/koinos-chain#843]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

I submitted a transaction with an rc_limit of `62853830` and the chain thrown the error `transaction reverted: insufficient rc`.
I put it in a for loop to try it again with more limit. In the next iteration the rc_limit was `162853830` and this time the transaction was accepted. The curious thing is that the `rc_used` was just `33055636`.

In summary,
```
rc_limit#1:  62853830 // failed with insufficient rc
rc_limit#2: 162853830 // OK
rc_used:     33055636
```

I don't understand why the first one failed knowing that it had enough rc_limit.

### Environment

```markdown
- OS:
```
